### PR TITLE
make AlignIO imports work without sqlite3

### PR DESCRIPTION
**Pull request by natechols**
- [ X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [ X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [ X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed.

This fixes a hard dependency on sqlite3 introduced in 4e77946e4c3c711e35d4a4ad40f52ae1dbc5e3bb.

### CODE REVIEW

File: Bio/AlignIO/MafIO.py [hunk 1]
```diff
@@ -35,7 +35,12 @@
 import os
 
 from itertools import islice
-from sqlite3 import dbapi2
+
+try:
+    from sqlite3 import dbapi2
+    from sqlite3.dbapi2 import OperationalError, DatabaseError
+except ImportError:
+    dbapi2 = OperationalError = DatabaseError = None
```
Comment (peterjc) [inline Bio/AlignIO/MafIO.py:43]: I don't think we need to define ``OperationalError`` and ``DatabaseError`` here.

Why not just revert the changes from 4e77946e4c3c711e35d4a4ad40f52ae1dbc5e3bb on this file?

**Comment by peterjc** [comment]
Any idea why your Python is missing sqlite3? Was it self-compiled and you missed some optional build time dependencies?

File: Bio/AlignIO/MafIO.py (hunk 1 above)
Comment (natechols) [reply Bio/AlignIO/MafIO.py:43]: That works for me too, I was just following the `MissingPythonDependencyError` used elsewhere.

**Comment by natechols** [comment]
@peterjc Exactly, we just never needed sqlite before.  Already fixed on our end but it cost us a build-test cycle so 🤷

**Comment by peterjc** [comment]
We could continue to bend of backwards to cope with sqlite3 being missing - things like the code here, or in the test suite:

https://github.com/biopython/biopython/blob/biopython-179/Bio/File.py

(and restore this behaviour in AlignIO)

Or we could remove all that special case code, and just say we require sqlite3. Much simpler, and with the rise of conda relatively few people are likely to have a self-compiled Python.

CC @mdehoon for comment.

**Comment by natechols** [comment]
If the sqlite3 dependency can be part of `setup.py` so we find out immediately, I would actually prefer that solution.

**Comment by peterjc** [comment]
Does that just work when ``sqlite3`` is normally in the standard library? Guess we can try it to find out...

**Comment by natechols** [comment]
Apparently not:
```
Searching for sqlite3
Reading https://pypi.org/simple/sqlite3/
Couldn't find index page for 'sqlite3' (maybe misspelled?)
Scanning index of all packages (this may take a while)
Reading https://pypi.org/simple/
No local packages or working download links found for sqlite3
error: Could not find suitable distribution for Requirement.parse('sqlite3')
```
So it would probably have to be `try/except ImportError` (and re-raise a custom error) in `setup.py`, I guess?

**Comment by peterjc** [comment]
Could you take a look at #369? This would resolve the issue by requiring ``sqlite3`` at installation time via ``setup.py`` and should remove all the special casing to cope without it.

**Comment by mdehoon** [comment]
> CC @mdehoon for comment.

Hopefully in the not too distant future, we can add a parser for bigMaf (self-indexed maf) files to Biopython. Then we will no longer need our own maf indexing code, so we won't need sqlite3, and the problem goes away.

**Comment by peterjc** [comment]
bigMaf support sounds good, but usage of sqlite3 will not go away - we use it in many places besides the maf indexing.

**Comment by mdehoon** [comment]
Still I would not require sqlite3 at compile time. It's not needed for most of Biopython, and installing it is not trivial.

**Comment by peterjc** [comment]
Why would you want to look at ``dbm`` rather than ``sqlite3`` at this point? Both are in the standard library and both are conditional on your Python being compiled with the necessary libraries present - see e.g. https://devguide.python.org/setup/#build-dependencies (search page for ``_dbm`` and ``_sqlite3``).

**Comment by mdehoon** [comment]
Sorry, I was under the impression that `dbm` did not have external dependencies. Never mind then.

**Comment by peterjc** [comment]
Are you happy for me to merge #3690 then?

i.e. drop all the code coping without sqlite3, and instead require it in ``setup.py``

**Comment by peterjc** [comment]
Conclusion from discussion on #3690 is to fix AlignIO to cope without sqlite3. i.e. The approach in this PR.

**Comment by peterjc** [comment]
Merging - thank you!
